lib/runtime-c-api/tests/test-exports.c

```diff
@@ -5,81 +5,483 @@
 
 int main()
 {
-    // Read the wasm file bytes
-    FILE *file = fopen("assets/sum.wasm", "r");
-    fseek(file, 0, SEEK_END);
-    long len = ftell(file);
-    uint8_t *bytes = malloc(len);
-    fseek(file, 0, SEEK_SET);
-    fread(bytes, 1, len, file);
-    fclose(file);
+    // Read the WebAssembly bytes.
+    uint8_t *wasm_bytes = NULL;
+    long wasm_bytes_length = 0;
+
+    {
+        FILE *file = fopen("assets/exports.wasm", "r");
+        fseek(file, 0, SEEK_END);
+        wasm_bytes_length = ftell(file);
+        wasm_bytes = (uint8_t *) malloc(wasm_bytes_length);
+        fseek(file, 0, SEEK_SET);
+        fread(wasm_bytes, 1, wasm_bytes_length, file);
+        fclose(file);
+    }
 
     wasmer_import_t imports[] = {};
     wasmer_instance_t *instance = NULL;
-    wasmer_result_t compile_result = wasmer_instantiate(&instance, bytes, len, imports, 0);
-    printf("Compile result: %d\n", compile_result);
+    wasmer_result_t compile_result = wasmer_instantiate(&instance, wasm_bytes, wasm_bytes_length, imports, 0);
+
     assert(compile_result == WASMER_OK);
 
     wasmer_exports_t *exports = NULL;
     wasmer_instance_exports(instance, &exports);
 
-    int exports_len = wasmer_exports_len(exports);
-    printf("exports_len: %d\n", exports_len);
-    assert(exports_len == 1);
+    int exports_length = wasmer_exports_len(exports);
+    printf("Number of exports: %d\n", exports_length);
+
+    {
+        printf("\nCheck the `sum` exported function\n");
+
+        wasmer_export_t *export = wasmer_exports_get(exports, 3);
+        wasmer_import_export_kind export_kind = wasmer_export_kind(export);
+
+        assert(export_kind == WASM_FUNCTION);
+
+        const wasmer_export_func_t *exported_function = wasmer_export_to_func(export);
+
+        wasmer_byte_array name_bytes = wasmer_export_name(export);
+
+        assert(name_bytes.bytes_len == 3);
+
+        char expected[] = {'s', 'u', 'm'};
+
+        printf("Read export name:\n");
+
+        for (uint32_t idx = 0; idx < name_bytes.bytes_len; idx++) {
+            printf("%c\n", name_bytes.bytes[idx]);
```

Comment:
Since we assert on it anyways, do we need this?

Reply:
We don't. The first example is just here to print some stuff, but that's pretty useless (it was taken fromt the previous version).